### PR TITLE
*: Update go-proxyprotocol version to fix some corner case

### LINKED ISSUE #43205: server: proxy protocol fallback feature will return header read timeout error when underlay connection read got error
## Bug Report

Please answer these questions before submitting your issue. Thanks!

server: proxy protocol fallback feature will return header read timeout error when underlay connection read got error 

### 1. Minimal reproduce step (Required)

```
start tidb with fallbackable=true
echo -ne "" | nc  localhost 4000 &> /dev/null
```

```
[2023/04/18 17:02:37.376 +08:00] [WARN] [server.go:612] ["Server.onConn handshake"] [conn=2199023255981] [error="Header read timeout"] ["remote addr"="[::1]:60100"]
```

### 2. What did you expect to see? (Required)

the error that `conn.Read` actually returned.

### 3. What did you see instead (Required)

got `Header read timeout` error.

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by blacktear23**
### What problem does this PR solve?

Issue Number: close #43205

Problem Summary:

go-proxyprotocol will return `ErrHeaderReadTimeout` error at parse header step with error returned `conn.Read` call. That will hide the original error object.

### What is changed and how it works?

update `go-proxyprotocol` package to new version will return the original error when enable fallback mode.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by blacktear23** [comment]
@xhebox PTAL

File: go.mod [hunk 1]
```diff
@@ -17,7 +17,7 @@ require (
 	github.com/ashanbrown/makezero v1.1.1
 	github.com/aws/aws-sdk-go v1.44.48
 	github.com/bazelbuild/buildtools v0.0.0-20230317132445-9c3c1fc0106e
-	github.com/blacktear23/go-proxyprotocol v1.0.5
```
Comment (hawkingrei) [inline go.mod:20]: Please run the ```make bazel_prepare```.

**Review by xhebox — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**